### PR TITLE
Docker image, Daemonset, Quickstart Guide

### PR DESCRIPTION
**Pull request by dougbtv**
Updates for a Docker image + Daemonset.

General idea is this is used to distribute Multus. The Docker image has instructions for compiling Multus. Using the Daemonset, host volumes from each node/master are exposed to the container as mounts, Multus binary & configuration are copied there. 

Also generates a kubeconfig using service account token auth, and includes RBAC for that service account.

Changes:

* Adds Docker image
* Has Travis changes to build & push Docker image
* Includes quickstart guide in the README.md
* Includes daemonset(s)

### CODE REVIEW

File: images/70-multus.conf [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+          {
+      "type": "flannel",
+      "name": "flannel.1",
+      "delegate": {
+        "isDefaultGateway": true
+      }
+    }
+  ],
+  "kubeconfig": "/etc/cni/net.d/multus.d/multus.kubeconfig"
+}
```
Comment (phoracek) [inline images/70-multus.conf]: missing newline

File: images/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      io.k8s.description="This is a component of OpenShift Container Platform and provides a meta CNI plugin." \
+      io.openshift.tags="openshift" \
+      maintainer="Doug Smith <dosmith@redhat.com>"
+
+ADD ./images/entrypoint.sh /
+
+# does it require a root user?
+# USER 1001
+
+ENTRYPOINT /entrypoint.sh
```
Comment (phoracek) [inline images/Dockerfile]: missing newline

File: images/entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+  fi
+
+  # Write a kubeconfig file for the CNI plugin.  Do this
+  # to skip TLS verification for now.  We should eventually support
+  # writing more complete kubeconfig files. This is only used
+  # if the provided CNI network config references it.
+  touch $MULTUS_KUBECONFIG
+  chmod ${KUBECONFIG_MODE:-600} $MULTUS_KUBECONFIG
+  cat > $MULTUS_KUBECONFIG <<EOF
+# Kubeconfig file for Calico CNI plugin.
```
Comment (squeed) [inline images/entrypoint.sh]: multus

File: images/70-multus.conf [hunk 4]
```diff
@@ -0,0 +1,14 @@
+{
+  "name": "multus-cni-network",
+  "type": "multus",
+  "delegates": [
+          {
```
Comment (phoracek) [inline images/70-multus.conf]: indentation does not match the other }

File: images/entrypoint.sh [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+else
+  echo "WARNING: Doesn't look like we're running in a kubernetes environment (no serviceaccount token)"
+fi
+
+# ---------------------- end Generate a "kube-config".
+
+echo "Entering sleep... (success)"
+
+# Sleep forever.
+sleep infinity
```
Comment (phoracek) [inline images/entrypoint.sh]: missing newline

**Comment by dougbtv** [comment]
Awesome, thanks for the feedback guys, I've got a new push -- including the correct nomenclature for Multus in there, plus the whitespace changes -- appreciate the eyes on it. Thanks again.

**Comment by dougbtv** [comment]
Noticed it's missing the CRD definition itself, need to add that.

**Comment by fepan** [comment]
The patch works fine on my local setup. Thanks Doug for the much improved deployment experience.

File: .travis.yml [hunk 6]
```diff
@@ -3,6 +3,10 @@ language: go
 # for the detail
 # sudo: requried
 dist: trusty
+env:
+  global:
+    - REGISTRY_USER=nfvperobot
```
Comment (s1061123) [inline .travis.yml]: How about to store env to travisCI GUI, not travis.yml?

**Comment by rkamudhan** [comment]
@fepan @s1061123  Can we merge this PR ?

File: .travis.yml (hunk 6 above)
Comment (s1061123) [reply .travis.yml]: Let's make another PR for that.

**Pull request merged**